frontend/openchat-client/src/openchat.ts

```diff
@@ -983,14 +979,17 @@ export class OpenChat extends EventTarget {
     }
 
     isPreviewing(chatId: string): boolean {
+        if (this.isReadOnly()) return true;
         return this.chatPredicate(chatId, isPreviewing);
     }
 
     isFrozen(chatId: string): boolean {
+        if (this.isReadOnly()) return true;
         return this.chatPredicate(chatId, isFrozen);
     }
 
     private chatPredicate(chatId: string, predicate: (chat: ChatSummary) => boolean): boolean {
+        if (this.isReadOnly()) return false;
```

Comment:
surely we can't do that? This is a completely generic function, we don't know what question is being asked of the chat.